### DISCUSSION TITLE
Running calculations on CityJSON model using QGIS

### DISCUSSION THREAD

**Discussion by mzdsmt** [cityjson/specs#134] [Q&A]
Hello, 
I am setting up a research project, and I am exploring how to use CityJSON, via QGIS, to extract data on polygons and especially run calculations on the visibility of building roofs.
I am having trouble understanding how to use this format for more than just visualization. I.e. QGIS does not support the 3D geometry, this seems to be just a rendering. A way around it would be to include the polygons coordinates on the data table, so I can operate with them, but I do not see how. For instance, I could find out and work with polygon centroids in x,y,z coordinates, normal vectors, etc.
Also, I could use a way of inheriting properties from underlying layers and use them to populate the polygons, but again, these do not seem like entities QGIS can operate with.
I am sure that I am missing something, so any hint to get me on track would be very much appreciated.
Thanks! Best,
M.

**Comment by hugoledoux** [comment]
As far as I know, QGIS is indeed mostly used for visualisation at this moment.

For what you want to do, I would advise use Python and extracting yourself the surfaces... or maybe try FME?

**Reply by mzdsmt** [reply]
Thank you... I had FME in my sights... but I was trying to avoid getting myself into yet another piece of software *roll up my sleeves...*
